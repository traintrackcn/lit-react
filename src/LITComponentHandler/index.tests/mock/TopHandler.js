import LITComponentHandler from "../..";
import p from '../../../__mock__/p';
import StateHandler from './StateHandler';
import ComponentHandler from './ComponentHandler';
import CommonHandler from './CommonHandler';


export default class TopHandler extends LITComponentHandler {


    getChildStateHandlerWithArrayPath() {
        if (!this._state) {
            this._state = new StateHandler(p.t, this);
        }
        return this._state;
    }

    getChildStateHandlerWithObjectPath() {
        if(!this._state1) {
            this._state1 = new StateHandler(p, this);
        }
        return this._state1
    }

    

    getChildComponentHandler() {
        if (!this._component) {
            this._component = new ComponentHandler(this);
        }
        return this._component;
    }

    getChildHandler() {
        if (!this._common) {
            this._common = new CommonHandler(this);
        }
        return this._common;
    }

    getCustomHandler() {
        return 'ok';
    }

}