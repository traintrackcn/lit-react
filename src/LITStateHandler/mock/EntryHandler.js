import LITBaseHandler from "../../LITBaseHandler";
import LITStateHandlerV2 from '../index_v2';
import p from './p';
import s from '../../s';
import LITComponentHandler from "../../LITComponentHandler";

export default class EntryHandler extends LITComponentHandler{

    constructor(ctx) {
        super(ctx);
        
    }

    getStateStore() {
        return s;
    }

    getH() {
        if (!this._h) {
            this._h = new LITStateHandlerV2(p.t, this);
        }
        return this._h;
    }
}