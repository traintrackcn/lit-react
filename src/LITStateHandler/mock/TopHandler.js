import LITStateHandlerV2 from '../index_v2';
import p from './p';
import s from '../../s';

export default class TopHandler extends LITStateHandlerV2{

    constructor(ctx) {
        super(ctx);
        this.setStateStore(s);
    }

    // getStateStore() {
    //     return s;
    // }

    getH() {
        if (!this._h) {
            this._h = new LITStateHandlerV2(p.t, this);
        }
        return this._h;
    }
}