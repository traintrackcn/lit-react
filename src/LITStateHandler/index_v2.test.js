import EntryHandler from './mock/EntryHandler';
import TopHandler from './mock/TopHandler';

// h.setStateStore(s);
const topH = new TopHandler();
const h = topH.getH();


describe('test state handler', () => {

    it('test setCollection', () => {

        h.del();

        const collection = [
            'a', 'b', 'c'
        ];
        h.setCollection(collection);
        expect(h.get()).toMatchSnapshot();
        h.select(1);
        expect(h.getValue()).toMatchSnapshot();

        expect(h.get()).toMatchSnapshot();
    });

    it('test getPath', () => {
        const expected = '["r","t"]';
        expect(JSON.stringify(h.getPath())).toBe(expected);
    })

    it('test set_value', () => {

        h.del();

        const value = { 'key': 'value' }
        h.setValue(value);
        expect(h.get()).toMatchSnapshot();
        expect(h.getValue()).toMatchSnapshot();
    });


});



