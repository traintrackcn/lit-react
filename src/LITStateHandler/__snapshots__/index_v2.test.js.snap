// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test state handler test set_value 1`] = `
Immutable.Map {
  "value": Immutable.Map {
    "key": "value",
  },
}
`;

exports[`test state handler test set_value 2`] = `
Immutable.Map {
  "key": "value",
}
`;

exports[`test state handler test setCollection 1`] = `
Immutable.Map {
  "collection": Immutable.List [
    "a",
    "b",
    "c",
  ],
}
`;

exports[`test state handler test setCollection 2`] = `"b"`;

exports[`test state handler test setCollection 3`] = `
Immutable.Map {
  "collection": Immutable.List [
    "a",
    "b",
    "c",
  ],
  "index": 1,
  "value": "b",
}
`;
